modules/imgproc/src/hal_replacement.hpp

```diff
@@ -0,0 +1,24 @@
+#ifndef OPENCV_IMGPROC_HAL_REPLACEMENT_HPP
+#define OPENCV_IMGPROC_HAL_REPLACEMENT_HPP
+
+#include "opencv2/core/hal/interface.h"
+
+inline int hal_ni_filterInit(void **, uchar *, size_t, int, int, int, int, int, int, int, int, double, int, int, bool, bool) { return CV_HAL_ERROR_NOT_IMPLEMENTED; }
```

Comment:
I would use struct cvhalFilter2D { ... }; and then passed pointer/double pointer to it to other functions, so that we have at least some compile-time control.